contracts/utils/math/Math.sol

```diff
@@ -537,41 +537,82 @@ library Math {
      * @dev Return the log in base 2 of a positive value rounded towards zero.
      * Returns 0 if given 0.
      */
-    function log2(uint256 value) internal pure returns (uint256) {
-        uint256 result = 0;
-        uint256 exp;
+    function log2(uint256 x) internal pure returns (uint256) {
+        // Efficient branchless algorithm for compute floor(log2(x)), the algorithm works as follow:
+        //
+        // 1. First round down `x` to the closest power of 2 using an modified version of the Seander's `Round up
+        //    to the next power of 2` algorithm, the version used here is modified to round down instead of up.
+        //    ref: https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+        //
+        // 2. Compute `n ≡ x mod 255`, we use this `n` value to split the equation in two parts that can be more
```

Comment:
Usually, notation `x ≡ y mod n` is used with x being anything, and y being between 0 (included) and n (excluded). 
I believe it would be valid to say that about any values x and y, provided that the residue modulo n is the same.

In that case, I think we are explicitelly getting that residue, so I would write it as 
```suggestion
        // 2. Compute `n`, the residue of `x` modulo 255. We use this `n` value to split the equation in two parts that can be more
```
or
```suggestion
        // 2. Compute `n`, the remainder of the division of `x` by 255. We use this `n` value to split the equation in two parts that can be more
```

Reply:
alternative would be to use `=` instead of `≡`, which would be more like a programming assignment (right to left) than a mathematical equivalent check.

Reply:
You are right, was using this symbol wrongly,  I decide to use `n = x mod 255` once seems more readable and intuitive for developers.